tests/integration/targets/test_mysql_user/tasks/test_privs_issue_465.yml

```
@@ -0,0 +1,53 @@
+# test code for privileges for mysql_user module - issue 465
+# (c) 2014,  Wayne Rosario <wrosario@ansible.com>
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+- vars:
+    mysql_parameters: &mysql_params
+      login_user: '{{ mysql_user }}'
+      login_password: '{{ mysql_password }}'
+      login_host: 127.0.0.1
+      login_port: '{{ mysql_primary_port }}'
+
+  block:
+
+    # ============================================================
+    - name: create admin user with ALL privs, without GRANT OPTIONS
+      mysql_user:
+        <<: *mysql_params
+        name: '{{ user_name_1 }}'
+        password: '{{ user_password_1 }}'
+        priv: '*.{{ db_name }}:ALL'
+        state: present
+
+    - include: assert_user.yml user_name={{user_name_2}} priv='ALL'
+
+    - name: create consumer user with all privileges using admin user
+      mysql_user:
+        <<: *mysql_params
+        name: '{{ user_name_2 }}'
+        password: '{{ user_password_2 }}'
+        login_user: '{{ user_name_1 }}'
+        login_password: '{{ user_password_1 }}'
+        priv: '*.{{ db_name }}:ALL'
+        state: present
+      register: result
```

```suggestion
        login_password: '{{ user_password_1 }}'
        priv: '*.{{ db_name }}:ALL'
        state: present
      register: result
```
the tests are green. You can put wrong params to generate an error. also put `ignore_errors: yes` to the task, then you can check `result.msg = "..."` in the following assert

---

if there are unclear things, please let me know

---

to be fair this is not ready to merge, as per previous messages i had troubles running ansible-test and managed to fix some of them. I can do the sanity tests but i cannot run ansible-test inside docker. I can run it with the virtual interpreters in my machine but I am not familiar with python so much or with this framework. I need to take a further look to make sure these test do reproduce what this change is about.

I will probably set a flag as work in progress as opposed to ready to merge.

---

@bizmate SGTM about putting WIP
i usually follow the following WF when testing:
```
# in my user's home dir
mkdir -p ansible_collections/community
git clone https://github.com/ansible-collections/community.mysql.git ansible_collections/community/mysql
# ^ you already have it, be sure the collection's root is in ansible_collections/community/mysql
git clone https://github.com/ansible/ansible
cd ansible
source hacking/env-setup
cd ansible_collections/community/mysql
# run integration tests
ansible-test integration test_mysql_user --docker -vvv > ~/test.log
# run sanity tests
ansible-test sanity [optional path to file] --docker
# run unit tests
ansible-test units [optional path to file] --docker
```
besides the [Quick-start guide](https://docs.ansible.com/ansible/devel/community/create_pr_quick_start.html)
we also have the [Integration tests guide](https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_integration_about.html)
Let me know if you have any questions / need help.

---

Also, I think it's fine to let Github run the tests for you. Look, It already did with 63 tests that all passed.
When testing locally, you must change manually the versions of Ansible, MySQL, Connector and Python. It's way easier to let GitHub Actions do its magic with the test matrix :)